### ISSUE TITLE
Multiple passes needed to get final formatting 

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/dart_style#1625]
Running `dart format` on the following code...

```dart
import 'dart:io';

bool foo() {
  if (
  // Explicitly stated to not be a bot.
  Platform.environment['BOT'] == 'false'
  ) {
    return false;
  }
  return true;
}
```

... produces this output:

```dart
import 'dart:io';

bool foo() {
  if (// Explicitly stated to not be a bot.
  Platform.environment['BOT'] == 'false') {
    return false;
  }
  return true;
}
```

Running `dort format` over that output again reformats it to the following (notice the added space before the `//`):

```dart
import 'dart:io';

bool foo() {
  if ( // Explicitly stated to not be a bot.
  Platform.environment['BOT'] == 'false') {
    return false;
  }
  return true;
}
```

Expectation: `dart format` produces the final output after one run. Subsequent runs do not change the output further.

**Comment by munificent** [comment]
Yeah, the formatter is intended to be idempotent. But, unfortunately, I've never been able to figure out a way to systematically *ensure* that it is in all cases. We do at least *test* now that all of the test cases are idempotent.

Most of the places where it fails to be have to do with comments in uncommon locations like here.

**Comment by RandalSchwartz** [comment]
> Most of the places where it fails to be have to do with comments in uncommon locations like here.

I presume you mean "uncommon" in the sense that most people never use comments. :)
